src/Bicep.LangServer/Handlers/BicepDocumentSymbolHandler.cs

```diff
@@ -52,21 +53,36 @@ private static DocumentSymbolRegistrationOptions GetSymbolRegistrationOptions()
 
         private IEnumerable<SymbolInformationOrDocumentSymbol> GetSymbols(CompilationContext context)
         {
-            return context.Compilation.GetEntrypointSemanticModel().Root.Declarations
-                .OrderBy(symbol=>symbol.DeclaringSyntax.Span.Position)
-                .Select(symbol => new SymbolInformationOrDocumentSymbol(CreateDocumentSymbol(symbol, context.LineStarts)));
+            var model = context.Compilation.GetEntrypointSemanticModel();
+            return model.Root.Declarations
+                .OrderBy(symbol => symbol.DeclaringSyntax.Span.Position)
+                .Select(symbol => new SymbolInformationOrDocumentSymbol(CreateDocumentSymbol(model, symbol, context.LineStarts)));
         }
 
-        private DocumentSymbol CreateDocumentSymbol(DeclaredSymbol symbol, ImmutableArray<int> lineStarts) =>
-            new DocumentSymbol
+        private DocumentSymbol CreateDocumentSymbol(SemanticModel model, DeclaredSymbol symbol, ImmutableArray<int> lineStarts)
+        {
+            var children = Enumerable.Empty<DocumentSymbol>();
+            if (symbol is ResourceSymbol resourceSymbol &&
+                resourceSymbol.DeclaringResource.TryGetBody() is ObjectSyntax body)
+            {
+                children = body.Resources
+                    .Select(r => model.GetSymbolInfo(r) as ResourceSymbol)
+                    .Where(s => s is ResourceSymbol)
+                    .Select(s => CreateDocumentSymbol(model, s!, lineStarts));
+            }
```

Comment:
@majastrz @anthony-c-martin - this works but seems non-ideal since it's pretty special-casey.

I think a better solution would be to make nested resource symbols children of the their containing resources. Then this code would look like:

```C#
children = body
    .Descendants.OfType<DeclaredSymbol>()
    .Select(s => CreateDocumentSymbol(model, s!, lineStarts));
```

I'm not sure what the other implication of that change would be, but it's one of the things I thought about and didn't do during the initial buildout of this feature.

LMK how you want to proceed. That change would be more involved, so I wanted to dicuss first.

Reply:
We had some discussions with @rynowak and @anthony-c-mart thin this. I think we're all in agreement that that approach is better. Unfortunately @rynowak won't be able to make these changes any time soon, so we decided to merge it as-is. 

Long term, we will revisit because we'd like to make the outline and  breadcrumbs richer by possibly adding properties and other nested elements.

Reply:
perhaps I could pick it up after I finish with loadFile?

Reply:
Sounds good to me. 👍

When you're ready to work on that, let's get on Teams and talk through the design so everyone is on the same page.